client/executor/common/src/sandbox.rs

```diff
@@ -236,21 +236,35 @@ impl<'a, FE: SandboxCapabilities + 'a> Externals for GuestExternals<'a, FE> {
 			.supervisor_externals
 			.allocate_memory(invoke_args_len)
 			.map_err(|_| trap("Can't allocate memory in supervisor for the arguments"))?;
-		self
+
+		let deallocate = |this: &mut GuestExternals<FE>, ptr, fail_msg| {
+			this
+				.supervisor_externals
+				.deallocate_memory(ptr)
+				.map_err(|_| trap(fail_msg))
+		};
+
+		if self
 			.supervisor_externals
 			.write_memory(invoke_args_ptr, &invoke_args_data)
-			.map_err(|_| trap("Can't write invoke args into memory"))?;
+			.is_err()
+		{
+			// Fixme: by returning here we're losing error context
```

Comment:
These "Fixme" should be removed.

We can just ignore the error of deallocate if we return an error anyway.